SU2_CFD/src/solvers/CMeshSolver.cpp

```diff
@@ -641,8 +654,15 @@ void CMeshSolver::ComputeGridVelocity(CGeometry **geometry, const CConfig *confi
       else if (secondOrder)
         GridVel = (1.5*Disp_nP1[iDim] - 2.0*Disp_n[iDim] + 0.5*Disp_nM1[iDim]) * invTimeStep;
 
-      geometry[MESH_0]->nodes->SetGridVel(iPoint, iDim, GridVel);
+      /*--- Keeping or overwriting any existing grid velocities ---*/
+      if (keep_existing_vel)
+    	geometry[MESH_0]->nodes->SetGridVel(iPoint, iDim, ExistingGridVel[iDim] + GridVel);
```

Comment:
... and here I do the superposition of the existing velocities with the deformation velocities.

Reply:
Yep but how do you configure SU2 to do both? I had the impression that the driver was not calling the old movement and new deformation.

Reply:
Honestly, I haven't fully understood the interaction of the different classes and in which sequence they are called, partially because they are distributed over several different files and also because of my poor C++ skills.


The CSinglezoneDriver::DynamicMeshUpdate first calls CIteration::SetGrid_Movement, then CIteration::SetMesh_Deformation.

I added a few new lines in CIteration::SetGrid_Movement, the switch/case statement for ROTATING_FRAME was already there, but unused.

Then, CIteration::SetMesh_Deformation deforms the elastic mesh and at a final stage, MeshSolver::ComputeGridVelocity is called, where I made sure the the existing, previously set velocities from the rotating frame are not overwritten. 

I believe that SU2 handles both cases combined now, although I'm not sure I did it right. That's the reason why I ask for help.